samples/besu/simpleasset/app/AssetExchangeERC20.js

```diff
@@ -0,0 +1,308 @@
+    await htlc.claimFungibleAsset(contractId, hashSecret, {
+      from: sender,
+    })
+
+    // Check tokens now owned by Alice
+    await assertTokenBal(
+      tokenContract,
+      sender,
+      tokenAmount,
+      `Alice doesn't not own ${tokenAmount} tokens`
+    )
+
+    const contractArr = await htlc.getContract.call(contractId)
+    const contract = htlcERC20ArrayToObj(contractArr)
+    assert.isTrue(contract.withdrawn) // withdrawn set
+    assert.isFalse(contract.refunded) // refunded still false
+    // with this the secret is out in the open and Bob will have knowledge of it
+    assert.equal(contract.preimage, hashSecret)
+
+    learnedSecret = contract.preimage
+}
+*/
+
+
+async function getBalances(Alice1, Bob1, Alice2, Bob2) {
+
+	var AliceAliceERC20Balance = await AliceERC20.balanceOf(Alice1)
+	console.log("Alice balance in AliceERC20 in Network 1", AliceAliceERC20Balance.toString())
+	
+	var AliceBobERC20Balance = await BobERC20.balanceOf(Alice2)
+	console.log("Alice balance in BobERC20 in Network 2", AliceBobERC20Balance.toString())
+	
+	var BobAliceERC20Balance = await AliceERC20.balanceOf(Bob1)
+	console.log("Bob balance in AliceERC20 in Network 1", await BobAliceERC20Balance.toString())
+	
+	var BobBobERC20Balance = await BobERC20.balanceOf(Bob2)
+	console.log("Bob balance in BobERC20 in Network 2", await BobBobERC20Balance.toString())
+
+}
+
+
+// Main entry point to the app
+async function main() {
+
+	// Network 1
+	const provider1 = new Web3.providers.HttpProvider('http://'+networkHost1+':'+networkPort1)
+	const web3N1 = new Web3(provider1)
+	const accounts1 = await web3N1.eth.getAccounts()
+
+	const Alice1 = accounts1[1] // owner of AliceERC20 and wants swap for BobERC20
+	const Bob1 = accounts1[2] // owner of BobERC20 and wants to swap for AliceERC20
+	console.log("Alice address in Network 1", Alice1)
+	console.log("Bob address in Network 1", Bob1)
+
+	const contractOwner1 = accounts1[0]
+
+
+	// Network 2
+	const provider2 = new Web3.providers.HttpProvider('http://'+networkHost2+':'+networkPort2)
+	const web3N2 = new Web3(provider2)
+	const accounts2 = await web3N2.eth.getAccounts()
+
+	const Alice2 = accounts2[1] // owner of AliceERC20 and wants swap for BobERC20
+	const Bob2 = accounts2[2] // owner of BobERC20 and wants to swap for AliceERC20
+	console.log("Alice address in Network 2", Alice2)
+	console.log("Bob address in Network 2", Bob2)
+
+	const contractOwner2 = accounts2[0]
+
+	// Initialization
+	const tokenSupply = 1000
+	const senderInitialBalance=100
+
+	await init(provider1, provider2, contractOwner1, contractOwner2, Alice1, Bob2, tokenSupply, senderInitialBalance)
+
+	console.log("\n Balances after init():")
+	await getBalances(Alice1, Bob1, Alice2, Bob2)
+
+    	hashPair = newSecretHashPair()
+	let tokenAmount=10
+	let timeOut = 15
+    	let timeLockSeconds = nowSeconds() + 2*timeOut
+
+	// Creating a HTLC contract for Alice locking her AliceERC20 tokens for Bob
```

Comment:
Indentations are all out of sync in the remainder of this code. Please align the statements (and comments).